backend/src/main/java/fk/prof/backend/worker/BackendDaemon.java

```diff
@@ -0,0 +1,198 @@
+
+import fk.prof.backend.ConfigManager;
+import fk.prof.backend.http.ApiPathConstants;
+import fk.prof.backend.http.ProfHttpClient;
+import fk.prof.backend.model.aggregation.AggregationWindowLookupStore;
+import fk.prof.backend.model.assignment.AggregationWindowPlannerStore;
+import fk.prof.backend.model.assignment.ProcessGroupAssociationStore;
+import fk.prof.backend.model.assignment.SimultaneousWorkAssignmentCounter;
+import fk.prof.backend.model.election.LeaderReadContext;
+import fk.prof.backend.proto.BackendDTO;
+import fk.prof.backend.util.ProtoUtil;
+import fk.prof.backend.util.proto.RecorderProtoUtil;
+import io.vertx.core.AbstractVerticle;
+import io.vertx.core.Future;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.core.json.JsonObject;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+import recording.Recorder;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+
+public class BackendDaemon extends AbstractVerticle {
+  private static Logger logger = LoggerFactory.getLogger(BackendDaemon.class);
+  private static String ENCODING = "UTF-8";
+
+  private final ConfigManager configManager;
+  private final LeaderReadContext leaderReadContext;
+  private final ProcessGroupAssociationStore processGroupAssociationStore;
+  private final SimultaneousWorkAssignmentCounter simultaneousWorkAssignmentCounter;
+  private final AggregationWindowLookupStore aggregationWindowLookupStore;
+  private final String ipAddress;
+  private final int leaderHttpPort;
+  private final int backendHttpPort;
+
+  private AggregationWindowPlannerStore aggregationWindowPlannerStore;
+  private ProfHttpClient httpClient;
+  private int loadTickCounter = 0;
+
+  public BackendDaemon(ConfigManager configManager,
+                       LeaderReadContext leaderReadContext,
+                       ProcessGroupAssociationStore processGroupAssociationStore,
+                       AggregationWindowLookupStore aggregationWindowLookupStore,
+                       SimultaneousWorkAssignmentCounter simultaneousWorkAssignmentCounter) {
+    this.configManager = configManager;
+    this.ipAddress = configManager.getIPAddress();
+    this.leaderHttpPort = configManager.getLeaderHttpPort();
+    this.backendHttpPort = configManager.getBackendHttpPort();
+
+    this.leaderReadContext = leaderReadContext;
+    this.processGroupAssociationStore = processGroupAssociationStore;
+    this.aggregationWindowLookupStore = aggregationWindowLookupStore;
+    this.simultaneousWorkAssignmentCounter = simultaneousWorkAssignmentCounter;
+  }
+
+  @Override
+  public void start() {
+    httpClient = buildHttpClient();
+    aggregationWindowPlannerStore = buildAggregationWindowPlannerStore();
+    postLoadToLeader();
+  }
+
+  private ProfHttpClient buildHttpClient() {
+    JsonObject httpClientConfig = configManager.getHttpClientConfig();
+    ProfHttpClient httpClient = ProfHttpClient.newBuilder()
```

Comment:
Fixed in commit https://github.com/Flipkart/fk-prof/commit/eaa57eb6a8bb821fd8543e80300e2965bbafcb28